plugins/destination/bigquery/client/client.go

```diff
@@ -0,0 +1,59 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/cloudquery/plugin-sdk/plugins"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+)
+
+type Client struct {
+	plugins.DefaultReverseTransformer
+	logger     zerolog.Logger
+	spec       specs.Destination
+	metrics    plugins.DestinationMetrics
+	client     *bigquery.Client
+	pluginSpec Spec
+	projectID  string
+	datasetID  string
+}
+
+func New(_ context.Context, logger zerolog.Logger, destSpec specs.Destination) (plugins.DestinationClient, error) {
+	if destSpec.WriteMode != specs.WriteModeAppend {
+		return nil, fmt.Errorf("bigquery destination only supports append mode")
+	}
+	c := &Client{
+		logger: logger.With().Str("module", "bq-dest").Logger(),
+	}
+	var spec Spec
+	c.spec = destSpec
+	if err := destSpec.UnmarshalSpec(&spec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal bigquery spec: %w", err)
+	}
+	spec.SetDefaults()
+	if err := spec.Validate(); err != nil {
+		return nil, err
+	}
+	client, err := bigquery.NewClient(context.Background(), spec.ProjectID)
```

Comment:
Using `ctx` doesn't work, as this context is used for all future queries (the `Query` command doesn't take a context). So by the time a query is made, the context has been cancelled. But maybe I should refactor this so that we create the client before every query, rather than at initialization?

Reply:
I've updated it so that a new client (with attached context) is created on every `Write` / `Migrate` call. I tested this and it seems to work fine.